bin/sozo/src/ops/events.rs

```diff
@@ -40,124 +39,455 @@ pub async fn execute(
 
     if let Some(events_map) = events_map {
         parse_and_print_events(res, events_map)?;
-    } else {
-        println!("{}", serde_json::to_string_pretty(&res)?);
     }
 
     Ok(())
 }
 
 fn parse_and_print_events(
     res: starknet::core::types::EventsPage,
-    events_map: HashMap<String, Vec<Event>>,
+    events_map: HashMap<String, Vec<Token>>,
 ) -> Result<()> {
     println!("Continuation token: {:?}", res.continuation_token);
     println!("----------------------------------------------");
     for event in res.events {
-        if let Some(e) = parse_event(event.clone(), &events_map) {
-            println!("{e}");
-        } else {
-            // Couldn't parse event
-            println!("{}", serde_json::to_string_pretty(&event)?);
+        match parse_event(event.clone(), &events_map) {
+            Ok(Some(e)) => {
+                println!("{e}");
+            }
+            Ok(None) => {
+                println!("No matching event found for {:?}", event);
+            }
+            Err(e) => {
+                println!("Error parsing event: {}", e);
+            }
         }
     }
     Ok(())
 }
 
+fn parse_core_basic(
+    cb: &CoreBasic,
+    value: &FieldElement,
+    include_felt_string: bool,
+) -> Result<String> {
+    match cb.type_name().as_str() {
+        "felt252" => {
+            let hex = format!("{:#x}", value);
+            match parse_cairo_short_string(value) {
+                Ok(parsed) if !parsed.is_empty() && include_felt_string => {
+                    Ok(format!("{} \"{}\"", parsed, hex))
+                }
+                _ => Ok(format!("\"{}\"", hex)),
```

Comment:
Wdyt about having the first always being `hex` without the double quote. And if it's parseable, we output the short string in double quotes?
```suggestion
                    Ok(format!("{} \"{}\"", hex, parsed))
                }
                _ => Ok(format!("{}", hex)),
```

Reply:
Good idea !